Pursuit-Core-iOS-Episodes-from-Online/Pursuit-Core-iOS-Episodes-from-Online/Controllers/ViewController.swift

```diff
@@ -0,0 +1,112 @@
+        @IBOutlet weak var showTableVIew: UITableView!
+        @IBOutlet weak var showSearchBar: UISearchBar!
+        @IBOutlet weak var spinnerOne: UIActivityIndicatorView!
+        
+        var viewShow = [TVShowsWrapper](){
+            didSet {
+                showTableVIew.reloadData()
+            }
+        }
+        
+        override func viewDidLoad() {
+            super.viewDidLoad()
+            showTableVIew.delegate = self
+            showTableVIew.dataSource = self
+            showSearchBar.delegate = self
+            loadData(word: nil)
+        }
+        var userSearchTerm: String? {
+            didSet {
+                self.showTableVIew.reloadData()
+            }
+        }
+        var filteredShow: [TVShowsWrapper]  {
+            guard let userSearchTerm = userSearchTerm else {
+                return viewShow
+            }
+            guard userSearchTerm != "" else {
+                return viewShow
+            }
+            return viewShow
+        }
+       
+
+      
+        override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+            guard let showVc = segue.destination as? EpisodesViewController else {
+                fatalError("Unexpected segue")
+            }
+            guard let selectedIndexPath = showTableVIew.indexPathForSelectedRow
+                else { fatalError("No row selected") }
+            showVc.show = filteredShow[selectedIndexPath.row]
+        }
+       private func loadData(word: String?){
+            TVShowsWrapper.getShow(userInput: word){ (result) in
+                switch result {
+                case .failure(let error):
+                    print(error)
+                case .success(let shows):
+                    DispatchQueue.main.async{
+                        return self.viewShow = shows
+                    }
+                }
+            }
+        }
+    }
+
+    extension ViewController: UITableViewDelegate, UITableViewDataSource{
+        func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+            return filteredShow.count
+        }
+        
+        func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+            let cell = showTableVIew.dequeueReusableCell(withIdentifier: "showListCell", for: indexPath) as? TVShowTableViewCell
+            let shows = filteredShow[indexPath.row].show
+            if let url = shows.image?.medium{
+                ImageHelper.shared.fetchImage(urlString: url) { (result) in
+                    DispatchQueue.main.async {
+                        switch result {
+                        case .failure(let error):
+                            print(error)
+                        case .success(let image):
+                             self.spinnerOne.stopAnimating()
+                            cell?.showImage.image = image
+                        }}}
+            }
+            cell?.showName.text = shows.name
```

Comment:
Good job handling the nil cases for the data! putting default values is a good idea when there is no data to be displayed.